packages/relayer/indexer/subscribe.go

```diff
@@ -63,3 +82,32 @@ func (svc *Service) subscribe(ctx context.Context, chainID *big.Int) error {
 		}
 	}
 }
+
+func (svc *Service) subscribeMessageStatusChanged(ctx context.Context, chainID *big.Int, errChan chan error) {
+	sink := make(chan *bridge.BridgeMessageStatusChanged)
+
+	sub := event.ResubscribeErr(svc.subscriptionBackoff, func(ctx context.Context, err error) (event.Subscription, error) {
+		if err != nil {
+			log.Errorf("svc.bridge.WatchMessageStatusChanged: %v", err)
+		}
+
+		return svc.bridge.WatchMessageStatusChanged(&bind.WatchOpts{
+			Context: ctx,
+		}, sink, nil)
+	})
+
+	defer sub.Unsubscribe()
+
+	for {
+		select {
+		case err := <-sub.Err():
```

Comment:
Do we need to also select `ctx.Done()` here?

Reply:
Will do!